float64 changeDirectionInterval
float64 startPosX
float64 startPosY
float64 startPosZ
float64 endPosX
float64 endPosY
float64 endPosZ
